coderd/coderdtest/oidctest/idp.go

```diff
@@ -1320,7 +1320,7 @@ func (f *FakeIDP) httpHandler(t testing.TB) http.Handler {
 // requests will fail.
 func (f *FakeIDP) HTTPClient(rest *http.Client) *http.Client {
 	if f.serve {
-		if rest == nil || rest.Transport == nil {
+		if rest == nil {
```

Comment:
I don't really understand why the `rest.Transport` check is being removed

Reply:
If it's there, the HTTP client in the function argument is basically **always** ignored. So it renders the `rest` client moot, and my cookie jar was not doing anything.

So I removed it, all tests still passed, so it should be safe since this is a testing pkg.

It also didn't feel correct. The idea is you can optionally pass in an `http.Client`. The transport being `nil` makes no sense to me :thinking: